wicker/core/column_files.py

```diff
@@ -85,22 +91,22 @@ class ColumnBytesFileWriter:
 
     def __init__(
         self,
-        storage: S3DataStorage = S3DataStorage(),
-        s3_path_factory: S3PathFactory = S3PathFactory(),
+        storage: AbstractDataStorage = S3DataStorage(),
+        path_factory: WickerPathFactory = S3PathFactory(),
         target_file_size: Optional[int] = None,
         target_file_rowgroup_size: Optional[int] = None,
         dataset_name: str = None,
     ) -> None:
         """Create a ColumnBytesFileWriter
 
         :param storage: storage client to use, defaults to S3DataStorage()
-        :param s3_path_factory: path factory to use, defaults to S3PathFactory()
+        :param path_factory: path factory to use, defaults to S3PathFactory()
         :param target_file_size: If set, open a new binary file when a column file gets larger than this many bytes.
         :param target_file_rowgroup_size: If set, open a new binary file when a column file has more rows than this
         :param dataset_name: dataset name, defaults to None
         """
         self.storage = storage
-        self.s3_path_factory = s3_path_factory
+        self.path_factory = path_factory
```

Comment:
@pickles-bread-and-butter could I ask you to add a couple of lines to the `__init__` here:
```
# Set an S3-specific reference to the path factory for backwards compatibility with previous versions of the class
self.s3_path_factory = path_factory if instance(path_factory, S3PathFactory) else None
```
I think this is pretty unnecessary, but Alborz has expressed a very strong concern that there could be a reference somewhere to an instance of the class that is referencing the `s3_path_factory` member variable. This should take care of it.

Reply:
Done